src/main/java/edu/illinois/cs/dt/tools/utility/iFixPlusPlugin.java

```diff
@@ -0,0 +1,744 @@
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void recordsubDiff(String beforeState,
+                               String afterState, String fileName) {
+        try {
+            // create a string builder
+            StringBuilder sb = new StringBuilder();
+            Diff diff = DiffBuilder.compare(beforeState).withTest(afterState).
+                    withNodeMatcher(new DefaultNodeMatcher(
+                            ElementSelectors.byName
+                    ))
+                    .checkForSimilar()
+                    .build();
+            Iterable<Difference> differences = diff.getDifferences();
+            for (Object object : differences) {
+                Difference difference = (Difference)object;
+
+                sb.append("***********************\n");
+                sb.append(difference);
+                sb.append("\n~~~~\n");
+                makeSubDifferenceReport(difference, sb);
+                sb.append("***********************\n");
+            }
+            writeToFile(fileName, sb.toString(), true);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public String readFile(String path) throws IOException {
+        File file = new File(path);
+        return FileUtils.readFileToString(file, "UTF-8");
+    }
+
+    Set<String> File2SetString(String path) {
+        File file = new File(path);
+        Set<String> keys = new HashSet<>();
+        try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+            String line;
+            while ((line = br.readLine()) != null) {
+                keys.add(line);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return keys;
+    }
+
+    private void diffSub() throws FileNotFoundException, UnsupportedEncodingException {
+        String subxml0 = subxmlFold + "/passing_order_xml";
+        String subxml1 = subxmlFold + "/failing_order_xml";
+        String afterRootPath= rootFold + "/failing_order.txt";
+        System.out.println(subxml0 + "; " + subxml1 + "; " + afterRootPath + "; ");
+        Set<String> afterRoots = File2SetString(afterRootPath);
+
+        System.out.println(diffFieldsFold + "; ");
+        for(String s: afterRoots) {
+
+            String path0 = subxml0 + "/" + s + ".xml";
+            String path1 = subxml1 + "/" + s + ".xml";
+            String state0 = ""; String state1 = "";
+            File file0 = new File(path0);
+            if (!file0.exists()) {
+                continue;
+            }
+            else {
+                try {
+                    state0 = readFile(path0);
+                    state1 = readFile(path1);
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                if (!state0.equals(state1)) {
+                    diffFields_filtered.add(s);
```

Comment:
Does this just compare whether the two strings read from the files are exactly the same? Shouldn't there be a more in-depth comparison to determine whether they are truly different?